sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineScaleSetImpl.java

```diff
@@ -1612,7 +1600,7 @@ protected void beforeCreating() {
                 virtualMachineScaleSet -> {
                     if (isManagedDiskEnabled()) {
                         this.managedDataDisks.setDataDisksDefaults();
-                    } else if (this.innerModel() != null && this.innerModel().virtualMachineProfile() != null) {
+                    } else {
                         List<VirtualMachineScaleSetDataDisk> dataDisks =
                             this.innerModel().virtualMachineProfile().storageProfile().dataDisks();
                         VirtualMachineScaleSetUnmanagedDataDiskImpl.setDataDisksDefaults(dataDisks, this.name());
```

Comment:
Any reason this check in `else` is removed? Seems `FLEXIBLE` still can get to this code?

Reply:
During current flow, profile can't be null either creating or updating vmss when executing to this point,